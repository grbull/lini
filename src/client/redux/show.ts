/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import { ShowWithEpisodesDto } from '../../server/show/show.dto';
import { api } from '../utils/api';

type ShowState = {
  status: 'init' | 'loading' | 'idle' | 'error';
  data?: ShowWithEpisodesDto;
  error?: string;
};

const initialState: ShowState = {
  status: 'init',
};

export const get = createAsyncThunk<ShowWithEpisodesDto, number>(
  'show/get',
  (id) => api.show.get(id)
);

export const show = createSlice({
  name: 'show',
  initialState,
  reducers: {
    reset: () => {
      return initialState;
    },
  },
  extraReducers: (builder) => {
    builder.addCase(get.pending, (state) => {
      state.status = 'loading';
      state.data = undefined;
      state.error = undefined;
    });
    builder.addCase(get.fulfilled, (state, { payload }) => {
      state.status = 'idle';
      state.data = payload;
    });
    builder.addCase(get.rejected, (state, { error }) => {
      state.status = 'error';
      state.error = error.message;
    });
  },
});

export const showActions = {
  ...show.actions,
  get,
};
