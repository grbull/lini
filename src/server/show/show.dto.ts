import { EpisodeDto } from '../episode/episode.dto';
import { NetworkDto } from '../network/network.dto';
import { DayOfWeek } from '../tvmaze/tvmaze.types';
import { WebChannelDto } from '../web_channel/web_channel.dto';

export class ShowDto {
  id!: number;
  name!: string;
  type!: string;
  language?: string;
  genres!: string[];
  status!: string;
  runtime!: number | null;
  runtimeAverage?: number;
  datePremiered?: string;
  officialSite?: string;
  scheduleTime!: string | null;
  scheduleDays!: (keyof typeof DayOfWeek)[];
  rating?: number;
  network?: NetworkDto;
  webChannel?: WebChannelDto;
  imageMedium?: string;
  imageOriginal?: string;
  summary?: string;
}

export class ShowWithEpisodesDto extends ShowDto {
  episodes!: EpisodeDto[];
}
