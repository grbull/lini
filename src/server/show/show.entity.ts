import { Exclude } from 'class-transformer';
import {
  Column,
  CreateDateColumn,
  Entity,
  ManyToOne,
  OneToMany,
  UpdateDateColumn,
} from 'typeorm';

import { EpisodeEntity } from '../episode/episode.entity';
import { NetworkEntity } from '../network/network.entity';
import { SubscriptionEntity } from '../subscription/subscription.entity';
import { DayOfWeek } from '../tvmaze/tvmaze.types';
import { WebChannelEntity } from '../web_channel/web_channel.entity';

@Entity('show')
export class ShowEntity {
  @Column('integer', { primary: true })
  id!: number;

  @Column('text')
  name!: string;

  // Scripted/Documentary/Reality/etc
  @Column('text')
  type!: string;

  @Column('text', { nullable: true })
  language!: string | null;

  @Column('text', { array: true })
  genres!: string[];

  // "Running or ended"
  @Column('text')
  status!: string;

  @Column('integer', { nullable: true })
  runtime!: number | null;

  @Column('integer', { nullable: true })
  runtimeAverage!: number | null;

  @Column('date', { nullable: true })
  datePremiered!: string | null;

  @Column('text', { nullable: true })
  officialSite!: string | null;

  @Column('time', { nullable: true })
  scheduleTime!: string | null;

  @Column('enum', { enum: DayOfWeek, array: true })
  scheduleDays!: (keyof typeof DayOfWeek)[];

  @Column('float', { nullable: true })
  rating!: number | null;

  @Column('integer')
  @Exclude()
  weight!: number;

  @ManyToOne(() => NetworkEntity, (network) => network.id, {
    nullable: true,
  })
  network!: NetworkEntity | null;

  @ManyToOne(() => WebChannelEntity, (webChannel) => webChannel.id, {
    nullable: true,
  })
  webChannel!: WebChannelEntity | null;

  @Column('integer', { nullable: true })
  @Exclude()
  tvrage!: number | null;

  @Column('integer', { nullable: true })
  @Exclude()
  thetvdb!: number | null;

  @Column('text', { nullable: true })
  @Exclude()
  imdb!: string | null;

  @Column('text', { nullable: true })
  imageMedium!: string | null;

  @Column('text', { nullable: true })
  imageOriginal!: string | null;

  @Column('text', { nullable: true })
  summary!: string | null;

  @OneToMany(() => EpisodeEntity, (episode) => episode.show)
  episodes!: EpisodeEntity[];

  @Column('bool', { default: false, select: false })
  @Exclude()
  isSeeded!: boolean;

  @OneToMany(() => SubscriptionEntity, (subscription) => subscription.show)
  @Exclude()
  subscriptions!: SubscriptionEntity[];

  @CreateDateColumn({ type: 'timestamptz' })
  @Exclude()
  dateCreated!: string;

  @UpdateDateColumn({ type: 'timestamptz' })
  @Exclude()
  dateUpdated!: string;
}
