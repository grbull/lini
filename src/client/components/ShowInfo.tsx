import cn from 'classnames';
import React, { ReactElement } from 'react';

import { ShowWithEpisodesDto } from '../../server/show/show.dto';
import { imageShow } from '../utils/imageShow';
import { EpisodeAccordion } from './EpisodeAccordion';
import { InfoCard } from './InfoCard';

interface Props {
  show: ShowWithEpisodesDto;
}

export function ShowInfo({ show }: Props): ReactElement {
  return (
    <>
      <InfoCard className={cn('my-0')}>
        <img
          alt={show.name}
          className={cn('w-1/3', 'float-left', 'mr-2.5')}
          src={imageShow(show.imageOriginal)}
        />
        <div className={cn()}>{show.summary}</div>
      </InfoCard>

      <InfoCard>
        <h2 className={cn('text-xl')}>Show Info</h2>
        <ul>
          <li className={cn('mt-2')}>
            <b>Network</b>: {show.network?.name ?? show.webChannel?.name}
          </li>
          <li className={cn('mt-0.5')}>
            <b>Schedule</b>:{' '}
            {show.scheduleDays.length > 0 ? (
              <span>
                {show.scheduleDays.map((day) => day + 's').join(', ')}
                {show.scheduleTime && (
                  <span> at {show.scheduleTime?.substr(0, 5)}</span>
                )}
                {show.runtime && <span> ({show.runtime} min)</span>}
              </span>
            ) : (
              'N/A'
            )}
          </li>
          <li className={cn('mt-0.5')}>
            <b>Status</b>: {show.status}
          </li>
          <li className={cn('mt-0.5')}>
            <b>Show type</b>: {show.type}
          </li>
          <li className={cn('mt-0.5')}>
            <b>Genres</b>: {show.genres.join(', ')}
          </li>
          <li className={cn('mt-3')}>
            <b>Official site</b>:{' '}
            {show.officialSite ? (
              <span>
                <a href={show.officialSite}>
                  {new URL(show.officialSite).hostname}
                </a>
              </span>
            ) : (
              'N/A'
            )}
          </li>

          <li className={cn('mt-3')}>
            <b>Rating</b>: {show.rating ?? 'N/A'}
          </li>
        </ul>
      </InfoCard>
      <InfoCard>
        <EpisodeAccordion episodes={show.episodes} />
      </InfoCard>
    </>
  );
}
