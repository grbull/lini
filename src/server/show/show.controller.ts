import {
  Controller,
  Get,
  Param,
  Query,
  UseGuards,
  UseInterceptors,
} from '@nestjs/common';

import { AuthGuard } from '../common/guards/auth.guard';
import { LoggingInterceptor } from '../common/interceptors/logging.interceptor';
import { ShowEntity } from './show.entity';
import { ShowService } from './show.service';

@Controller('show')
@UseGuards(AuthGuard)
@UseInterceptors(LoggingInterceptor)
export class ShowController {
  constructor(private readonly showService: ShowService) {}

  @Get('autocomplete')
  public autoComplete(@Query('q') query: string): Promise<ShowEntity[]> {
    return this.showService.autocomplete(query);
  }

  // TODO: Implement pagination
  @Get('search/:searchQuery')
  public search(
    @Param('searchQuery') searchQuery: string
  ): Promise<ShowEntity[]> {
    return this.showService.search(searchQuery);
  }

  @Get(':id')
  public getOne(@Param('id') id: number): Promise<ShowEntity> {
    return this.showService.getOne(id);
  }

  // public getShowsByLanguage() {}

  // public getShowsByNetwork() {}

  // public getShowsByWebChannel() {}
}
