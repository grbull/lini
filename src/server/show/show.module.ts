import { Module } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';

import { LoggerModule } from '../logger/logger.module';
import { NetworkModule } from '../network/network.module';
import { WebChannelModule } from '../web_channel/web_channel.module';
import { ShowController } from './show.controller';
import { ShowEntity } from './show.entity';
import { ShowService } from './show.service';

@Module({
  imports: [
    LoggerModule,
    NetworkModule,
    WebChannelModule,
    TypeOrmModule.forFeature([ShowEntity]),
  ],
  providers: [ShowService],
  controllers: [ShowController],
  exports: [ShowService],
})
export class ShowModule {}
