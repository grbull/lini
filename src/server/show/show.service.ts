import { Injectable, NotFoundException } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Raw, Repository } from 'typeorm';

import { LoggerService } from '../logger/logger.service';
import { NetworkService } from '../network/network.service';
import { TvMazeShow, TvMazeShowEmbedded } from '../tvmaze/tvmaze.types';
import { TvMazeShowToEntity } from '../tvmaze/tvmaze.utils';
import { WebChannelService } from '../web_channel/web_channel.service';
import { ShowEntity } from './show.entity';

@Injectable()
export class ShowService {
  constructor(
    private readonly loggerService: LoggerService,
    private readonly networkService: NetworkService,
    private readonly webChannelService: WebChannelService,
    @InjectRepository(ShowEntity)
    private readonly showRepository: Repository<ShowEntity>
  ) {
    this.loggerService.setContext('ShowService');
  }

  /**
   * Used for search and autocomplete when typing in a search box. Returns 5 items
   * Considering returning paginated result for search. TBD
   */
  public autocomplete(query: string): Promise<ShowEntity[]> {
    return this.showRepository
      .createQueryBuilder('show')
      .where({
        name: Raw((name) => `LOWER(${name}) LIKE '%${query.toLowerCase()}%'`),
      })
      .leftJoinAndSelect('show.network', 'network')
      .leftJoinAndSelect('network.country', 'networkCountry')
      .leftJoinAndSelect('show.webChannel', 'webChannel')
      .leftJoinAndSelect('webChannel.country', 'webChannelCountry')
      .take(5)
      .getMany();
  }

  public search(query: string): Promise<ShowEntity[]> {
    return this.showRepository
      .createQueryBuilder('show')
      .where({
        name: Raw((name) => `LOWER(${name}) LIKE '%${query.toLowerCase()}%'`),
      })
      .leftJoinAndSelect('show.network', 'network')
      .leftJoinAndSelect('network.country', 'networkCountry')
      .leftJoinAndSelect('show.webChannel', 'webChannel')
      .leftJoinAndSelect('webChannel.country', 'webChannelCountry')
      .take(48)
      .getMany();
  }

  public getOne(id: number): Promise<ShowEntity> {
    try {
      return this.showRepository
        .createQueryBuilder('show')
        .where('show.id = :id', { id })
        .leftJoinAndSelect('show.network', 'network')
        .leftJoinAndSelect('network.country', 'networkCountry')
        .leftJoinAndSelect('show.webChannel', 'webChannel')
        .leftJoinAndSelect('webChannel.country', 'webChannelCountry')
        .leftJoinAndSelect('show.episodes', 'episode')
        .orderBy('episode.airdate', 'ASC')
        .getOneOrFail();
    } catch {
      throw new NotFoundException();
    }
  }

  // All of the below is used primarily for seeding and updating from tvmaze
  private async createOneFromTvMaze(
    show: TvMazeShow | TvMazeShowEmbedded
  ): Promise<ShowEntity> {
    const webChannel = show.webChannel
      ? (await this.webChannelService.createOrUpdate(show.webChannel)).id
      : null;
    const network = show.network
      ? (await this.networkService.createOrUpdate(show.network)).id
      : null;

    return this.showRepository.create(
      TvMazeShowToEntity(show, network, webChannel)
    );
  }

  private async createManyFromTvMaze(
    shows: TvMazeShow[] | TvMazeShowEmbedded[]
  ): Promise<ShowEntity[]> {
    const showEntities: ShowEntity[] = [];

    // Must be done synchronously to avoid duplicate relations
    for (const show of shows) {
      showEntities.push(await this.createOneFromTvMaze(show));
    }
    return showEntities;
  }

  private async filterExisting(shows: TvMazeShow[]): Promise<TvMazeShow[]> {
    const existingShows = await this.showRepository.find({
      select: ['id'],
      where: shows.map((show) => Object({ id: show.id })),
    });

    return shows.filter(
      (show) =>
        !existingShows.find((existingShow) => existingShow.id === show.id)
    );
  }

  /**
   * Used when seeding series. We filter out existing to avoid altering
   * the update column. We want to avoid marking series as updated, when we
   * havn't updated the relevant episodes.
   */
  public async createManyIfDoesNotExist(
    shows: TvMazeShow[]
  ): Promise<ShowEntity[]> {
    const showsToCreate = await this.filterExisting(shows);
    const showEntities = await this.createManyFromTvMaze(showsToCreate);

    return this.showRepository.save(showEntities);
  }

  public async deleteOneById(showId: number): Promise<void> {
    await this.showRepository.delete(showId);
  }

  public async updateShow(show: TvMazeShowEmbedded): Promise<ShowEntity> {
    const showEntity = await this.createOneFromTvMaze(show);
    return this.showRepository.save(showEntity);
  }

  public async markSeeded(showId: number): Promise<void> {
    await this.showRepository.update(showId, { isSeeded: true });
  }

  public getShowsToSeed(): Promise<
    Pick<ShowEntity, 'id' | 'isSeeded' | 'name'>[]
  > {
    return this.showRepository
      .createQueryBuilder('show')
      .select(['show.id', 'show.isSeeded', 'show.name'])
      .where('show.isSeeded = :isSeeded', { isSeeded: false })
      .orderBy('show.id', 'ASC')
      .getMany();
  }

  public getShowsLastUpdated(): Promise<
    Pick<ShowEntity, 'id' | 'dateUpdated'>[]
  > {
    return this.showRepository
      .createQueryBuilder('show')
      .select(['show.id', 'show.dateUpdated'])
      .orderBy('show.id', 'ASC')
      .getMany();
  }
}
