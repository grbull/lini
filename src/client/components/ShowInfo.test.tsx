/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import React from 'react';

import { ShowWithEpisodesDto } from '../../server/show/show.dto';
import { ShowInfo } from './ShowInfo';

const showDto: ShowWithEpisodesDto = {
  id: 41007,
  name: 'Loki',
  type: 'Scripted',
  language: 'English',
  genres: ['Action', 'Adventure', 'Science-Fiction'],
  status: 'Running',
  runtime: null,
  runtimeAverage: 48,
  datePremiered: '2021-06-09',
  officialSite: 'https://www.marvel.com/tv-shows/loki/1',
  scheduleTime: null,
  scheduleDays: ['Wednesday'],
  rating: 8.2,
  imageMedium: '/uploads/images/medium_portrait/320/801227.jpg',
  imageOriginal: '/uploads/images/original_untouched/320/801227.jpg',
  summary:
    'Loki follows the trickster and shape-shifter god who pops up throughout human history as an unlikely influencer on historical events.',
  network: null,
  webChannel: { id: 287, name: 'Disney+', country: null },
  episodes: [
    {
      id: '1984092',
      name: 'Glorious Purpose',
      season: 1,
      number: 1,
      type: 'regular',
      airdate: '2021-06-09',
      airtime: null,
      airstamp: '2021-06-09T12:00:00.000Z',
      runtime: 50,
      imageMedium: '/uploads/images/medium_landscape/327/817543.jpg',
      imageOriginal: '/uploads/images/original_untouched/327/817543.jpg',
      summary:
        "Loki, the God of Mischief, finds himself out of time and in an unusual place and forced — against his godly disposition — to cooperate with others. Welcome to the TVA, y'all!",
    },
    {
      id: '2095474',
      name: 'The Variant',
      season: 1,
      number: 2,
      type: 'regular',
      airdate: '2021-06-16',
      airtime: null,
      airstamp: '2021-06-16T12:00:00.000Z',
      runtime: 53,
      imageMedium: '/uploads/images/medium_landscape/329/823863.jpg',
      imageOriginal: '/uploads/images/original_untouched/329/823863.jpg',
      summary:
        'Mobius puts Loki to work investigating The Variant, but not everyone at the TVA is thrilled to be working alongside The God of Mischief.',
    },
    {
      id: '2095475',
      name: 'Lamentis',
      season: 1,
      number: 3,
      type: 'regular',
      airdate: '2021-06-23',
      airtime: null,
      airstamp: '2021-06-23T12:00:00.000Z',
      runtime: 41,
      imageMedium: '/uploads/images/medium_landscape/332/831232.jpg',
      imageOriginal: '/uploads/images/original_untouched/332/831232.jpg',
      summary:
        'Loki finds himself trapped in an apocalypse with a dead TemPad and the last person he wants by his side: the Loki Variant, Sylvie. Now the two must find their way off of the doomed moon before complete annihilation.',
    },
    {
      id: '2095476',
      name: 'The Nexus Event',
      season: 1,
      number: 4,
      type: 'regular',
      airdate: '2021-06-30',
      airtime: null,
      airstamp: '2021-06-30T12:00:00.000Z',
      runtime: 48,
      imageMedium: '/uploads/images/medium_landscape/334/836858.jpg',
      imageOriginal: '/uploads/images/original_untouched/334/836858.jpg',
      summary:
        "Loki and Sylvie find their way off the Lamentis-1 mere moments before complete annihilation. Now, the question on everyone's mind, especially those in charge of the TVA — how did these two variants create a Nexus Event in the face of an apocalypse?",
    },
    {
      id: '2095477',
      name: 'Journey Into Mystery',
      season: 1,
      number: 5,
      type: 'regular',
      airdate: '2021-07-07',
      airtime: null,
      airstamp: '2021-07-07T12:00:00.000Z',
      runtime: 48,
      imageMedium: '/uploads/images/medium_landscape/337/843491.jpg',
      imageOriginal: '/uploads/images/original_untouched/337/843491.jpg',
      summary:
        "Upon discovering the Time Keepers aren't real, Judge Renslayer prunes Loki with the Time Stick. Now in an unknown land surrounded by variant Lokis, they urge him to follow them. Meanwhile, turning on the judge, Sylvie demands to be told everything about the TVA.",
    },
    {
      id: '2095478',
      name: 'Episode 6',
      season: 1,
      number: 6,
      type: 'regular',
      airdate: '2021-07-14',
      airtime: null,
      airstamp: '2021-07-14T12:00:00.000Z',
      runtime: null,
      imageMedium: null,
      imageOriginal: null,
      summary: null,
    },
  ],
};

function setup(show: ShowWithEpisodesDto): RenderResult {
  const utils = render(<ShowInfo show={show} />);

  return { ...utils };
}

describe('ShowInfo Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup(showDto);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot with null values', () => {
    const { asFragment } = setup({
      ...showDto,
      network: null,
      webChannel: null,
      scheduleDays: [],
      scheduleTime: null,
      runtime: null,
      officialSite: null,
      rating: null,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot with network value', () => {
    const { asFragment } = setup({
      ...showDto,
      network: {
        id: 1,
        name: 'Test network',
        country: {
          code: 'UK',
          name: 'United Kingdom',
          timezone: 'London/Europe',
        },
      },
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot with a schedule', () => {
    const { asFragment } = setup({
      ...showDto,
      scheduleTime: '20:00',
      runtime: 60,
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
