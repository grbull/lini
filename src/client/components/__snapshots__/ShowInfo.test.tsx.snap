// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowInfo Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden my-0"
  >
    <div
      class="w-1/3 mr-2.5 float-left"
    >
      <div
        class="h-0 relative"
        style="padding-top: 140.47619047619045%;"
      >
        <div
          class="animate-pulse bg-gray-700 absolute top-0 left-0 w-full h-full"
        />
        <img
          alt="Loki"
          class="absolute top-0 left-0 w-full h-full hidden"
          src="undefined/uploads/images/medium_portrait/320/801227.jpg"
        />
      </div>
    </div>
    <div
      class=""
    >
      Loki follows the trickster and shape-shifter god who pops up throughout human history as an unlikely influencer on historical events.
    </div>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <h2
      class="text-xl"
    >
      Show Info
    </h2>
    <ul>
      <li
        class="mt-2"
      >
        <b>
          Network
        </b>
        : Disney+
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Schedule
        </b>
        : 
        <span>
          Wednesdays
        </span>
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Status
        </b>
        : Running
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Show type
        </b>
        : Scripted
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Genres
        </b>
        : Action, Adventure, Science-Fiction
      </li>
      <li
        class="mt-3"
      >
        <b>
          Official site
        </b>
        : 
        <span>
          <a
            href="https://www.marvel.com/tv-shows/loki/1"
          >
            www.marvel.com
          </a>
        </span>
      </li>
      <li
        class="mt-3"
      >
        <b>
          Rating
        </b>
        : 8.2
      </li>
    </ul>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <h2
      class="text-xl mb-2"
    >
      Episode List
    </h2>
    <div
      class="py-1"
      role="button"
      tabindex="1"
    >
      <div
        class="flex justify-between items-center"
      >
        <span>
          Season 1
        </span>
        <span>
          6 Episodes
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShowInfo Component matches the snapshot with a schedule 1`] = `
<DocumentFragment>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden my-0"
  >
    <div
      class="w-1/3 mr-2.5 float-left"
    >
      <div
        class="h-0 relative"
        style="padding-top: 140.47619047619045%;"
      >
        <div
          class="animate-pulse bg-gray-700 absolute top-0 left-0 w-full h-full"
        />
        <img
          alt="Loki"
          class="absolute top-0 left-0 w-full h-full hidden"
          src="undefined/uploads/images/medium_portrait/320/801227.jpg"
        />
      </div>
    </div>
    <div
      class=""
    >
      Loki follows the trickster and shape-shifter god who pops up throughout human history as an unlikely influencer on historical events.
    </div>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <h2
      class="text-xl"
    >
      Show Info
    </h2>
    <ul>
      <li
        class="mt-2"
      >
        <b>
          Network
        </b>
        : Disney+
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Schedule
        </b>
        : 
        <span>
          Wednesdays
          <span>
             at 20:00
          </span>
          <span>
             (60 min)
          </span>
        </span>
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Status
        </b>
        : Running
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Show type
        </b>
        : Scripted
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Genres
        </b>
        : Action, Adventure, Science-Fiction
      </li>
      <li
        class="mt-3"
      >
        <b>
          Official site
        </b>
        : 
        <span>
          <a
            href="https://www.marvel.com/tv-shows/loki/1"
          >
            www.marvel.com
          </a>
        </span>
      </li>
      <li
        class="mt-3"
      >
        <b>
          Rating
        </b>
        : 8.2
      </li>
    </ul>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <h2
      class="text-xl mb-2"
    >
      Episode List
    </h2>
    <div
      class="py-1"
      role="button"
      tabindex="1"
    >
      <div
        class="flex justify-between items-center"
      >
        <span>
          Season 1
        </span>
        <span>
          6 Episodes
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShowInfo Component matches the snapshot with network value 1`] = `
<DocumentFragment>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden my-0"
  >
    <div
      class="w-1/3 mr-2.5 float-left"
    >
      <div
        class="h-0 relative"
        style="padding-top: 140.47619047619045%;"
      >
        <div
          class="animate-pulse bg-gray-700 absolute top-0 left-0 w-full h-full"
        />
        <img
          alt="Loki"
          class="absolute top-0 left-0 w-full h-full hidden"
          src="undefined/uploads/images/medium_portrait/320/801227.jpg"
        />
      </div>
    </div>
    <div
      class=""
    >
      Loki follows the trickster and shape-shifter god who pops up throughout human history as an unlikely influencer on historical events.
    </div>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <h2
      class="text-xl"
    >
      Show Info
    </h2>
    <ul>
      <li
        class="mt-2"
      >
        <b>
          Network
        </b>
        : Test network
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Schedule
        </b>
        : 
        <span>
          Wednesdays
        </span>
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Status
        </b>
        : Running
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Show type
        </b>
        : Scripted
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Genres
        </b>
        : Action, Adventure, Science-Fiction
      </li>
      <li
        class="mt-3"
      >
        <b>
          Official site
        </b>
        : 
        <span>
          <a
            href="https://www.marvel.com/tv-shows/loki/1"
          >
            www.marvel.com
          </a>
        </span>
      </li>
      <li
        class="mt-3"
      >
        <b>
          Rating
        </b>
        : 8.2
      </li>
    </ul>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <h2
      class="text-xl mb-2"
    >
      Episode List
    </h2>
    <div
      class="py-1"
      role="button"
      tabindex="1"
    >
      <div
        class="flex justify-between items-center"
      >
        <span>
          Season 1
        </span>
        <span>
          6 Episodes
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShowInfo Component matches the snapshot with null values 1`] = `
<DocumentFragment>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden my-0"
  >
    <div
      class="w-1/3 mr-2.5 float-left"
    >
      <div
        class="h-0 relative"
        style="padding-top: 140.47619047619045%;"
      >
        <div
          class="animate-pulse bg-gray-700 absolute top-0 left-0 w-full h-full"
        />
        <img
          alt="Loki"
          class="absolute top-0 left-0 w-full h-full hidden"
          src="undefined/uploads/images/medium_portrait/320/801227.jpg"
        />
      </div>
    </div>
    <div
      class=""
    >
      Loki follows the trickster and shape-shifter god who pops up throughout human history as an unlikely influencer on historical events.
    </div>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <h2
      class="text-xl"
    >
      Show Info
    </h2>
    <ul>
      <li
        class="mt-2"
      >
        <b>
          Network
        </b>
        : N/A
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Schedule
        </b>
        : N/A
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Status
        </b>
        : Running
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Show type
        </b>
        : Scripted
      </li>
      <li
        class="mt-0.5"
      >
        <b>
          Genres
        </b>
        : Action, Adventure, Science-Fiction
      </li>
      <li
        class="mt-3"
      >
        <b>
          Official site
        </b>
        : N/A
      </li>
      <li
        class="mt-3"
      >
        <b>
          Rating
        </b>
        : N/A
      </li>
    </ul>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <h2
      class="text-xl mb-2"
    >
      Episode List
    </h2>
    <div
      class="py-1"
      role="button"
      tabindex="1"
    >
      <div
        class="flex justify-between items-center"
      >
        <span>
          Season 1
        </span>
        <span>
          6 Episodes
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
